geomesa-cassandra/geomesa-cassandra-datastore/src/main/scala/org/locationtech/geomesa/cassandra/data/CassandraDataStore.scala

```diff
@@ -0,0 +1,147 @@
+  * are made available under the terms of the Apache License, Version 2.0
+  * which accompanies this distribution and is available at
+  * http://www.opensource.org/licenses/apache2.0.php.
+  *************************************************************************/
+
+package org.locationtech.geomesa.cassandra.data
+
+import java.math.BigInteger
+import java.net.URI
+import java.nio.ByteBuffer
+import java.util
+import java.util.{Date, UUID}
+
+import com.datastax.driver.core._
+import com.google.common.collect.HashBiMap
+import com.vividsolutions.jts.geom.{Geometry, Point}
+import org.geotools.data.store._
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder
+import org.geotools.feature.{AttributeTypeBuilder, NameImpl}
+import org.joda.time.{DateTime, Seconds, Weeks}
+import org.locationtech.geomesa.curve.Z3SFC
+import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType
+import org.locationtech.geomesa.utils.text.WKBUtils
+import org.locationtech.sfcurve.zorder.ZCurve2D
+import org.opengis.feature.`type`.{AttributeDescriptor, Name}
+import org.opengis.feature.simple.SimpleFeatureType
+
+object CassandraDataStore {
+  import scala.collection.JavaConversions._
+
+  val typeMap = HashBiMap.create[Class[_], DataType]
+  typeMap.putAll(Map(
+    classOf[Integer]    -> DataType.cint(),
+    classOf[Long]       -> DataType.bigint(),
+    classOf[Float]      -> DataType.cfloat(),
+    classOf[Double]     -> DataType.cdouble(),
+    classOf[Boolean]    -> DataType.cboolean(),
+    classOf[BigDecimal] -> DataType.decimal(),
+    classOf[BigInteger] -> DataType.varint(),
+    classOf[String]     -> DataType.text(),
+    classOf[Date]       -> DataType.timestamp(),
+    classOf[UUID]       -> DataType.uuid(),
+    classOf[Point]      -> DataType.blob()
+  ))
+
+  def getSchema(name: Name, table: TableMetadata): SimpleFeatureType = {
+    val cols = table.getColumns.filterNot { c => c.getName == "pkz" || c.getName == "z31" || c.getName == "fid" }
+    val attrTypeBuilder = new AttributeTypeBuilder()
+    val attributes = cols.map { c =>
+      val it = typeMap.inverse().get(c.getType)
+      attrTypeBuilder.binding(it).buildDescriptor(c.getName)
+    }
+    // TODO: allow user data to set dtg field
+    val dtgAttribute = attributes.find(_.getType.getBinding.isAssignableFrom(classOf[java.util.Date])).head
+    val sftBuilder = new SimpleFeatureTypeBuilder()
+    sftBuilder.addAll(attributes)
+    sftBuilder.setName(name.getLocalPart)
+    val sft = sftBuilder.buildFeatureType()
+    sft.getUserData.put(RichSimpleFeatureType.DEFAULT_DATE_KEY, dtgAttribute.getLocalName)
+    sft
+  }
+
+  sealed trait FieldSerializer {
+    def serialize(o: java.lang.Object): java.lang.Object
+    def deserialize(o: java.lang.Object): java.lang.Object
+  }
+  case object GeomSerializer extends FieldSerializer {
+    override def serialize(o: Object): AnyRef = {
+      val geom = o.asInstanceOf[Point]
+      ByteBuffer.wrap(WKBUtils.write(geom))
```

Comment:
might be worthwhile to use kryo encoding here:
https://github.com/locationtech/geomesa/blob/master/geomesa-features/geomesa-feature-kryo/src/test/scala/org/locationtech/geomesa/features/kryo/KryoGeometrySerializerTest.scala#L39-L40